### ISSUE TITLE
bar renders on top of exwm's workspace frame

### ISSUE THREAD

**Issue by agzam** [ch11ng/exwm#432]
I can't find a way to fix this thing, I'm not even sure if this is a bug. I hope someone can share some ideas.

I'm using yabar and at the beginning of the session it works fine - yabar and Emacs frame are not overlapping, but as soon I connect/disconnect external display or change any xrandr options the bar overlays on top of the Emacs frame.

**upd:** It seems whenever exwm ads a new workspace it doesn't take into account anything that's on the screen and just makes the frame occupy entire screen

**Comment by agzam** [comment]
this is what xprop returns for yabar:

    xprop
    WM_NAME(STRING) = "yabar"
    _NET_WM_STRUT(CARDINAL) = 0, 0, 0, 22
    _NET_WM_STRUT_PARTIAL(CARDINAL) = 0, 0, 0, 22, 0, 0, 0, 0, 0, 0, 0, 1919
    _NET_WM_DESKTOP(CARDINAL) = 4294967295
    _NET_WM_STATE(ATOM) = _NET_WM_STATE_STICKY, _NET_WM_STATE_ABOVE
    _NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_DOCK

**Comment by ch11ng** [comment]
At least the strut set is correct.  What's the value of `exwm-workspace--struts` and `exwm-workspace--workareas`.  And what if you manually invoke `(exwm-workspace--update-workareas)`?

**Comment by agzam** [comment]
the value of `exwm-workspace--struts` is `([bottom 22 [1920 4479]])`. calling `(exwm-workspace--update-workareas)` or `(exwm-workspace--update-struts)` didn't change anything for me

**Comment by agzam** [comment]
ha... surprisingly this problem only appears when I configure yabar to be in the bottom. When it's forced to be on top - it doesn't seem to be a problem. I guess maybe this has nothing to do with EXWM.

**Comment by ch11ng** [comment]
What's the value of `exwm-workspace--workareas`? It sounds like a bug in EXWM.

**Comment by agzam** [comment]
with the bar in the bottom, overlapping on top of the exwm workspace frame:
Its value is `([0 1264 1920 1080] [1920 0 2560 1440])`

**Comment by agzam** [comment]
damn... now I'm seeing it overlapping when the bar is at the top. I guess it is a bug in exwm ;(

**Comment by ch11ng** [comment]
The value of `exwm-workspace--workareas` is quite strange.  What's the setting of RandR? It seems there are two monitors side by side with the smaller one on the left, but they are not top/bottom aligned.  I'm not sure if it's intended but in this case the 'strut' should be set (by yabar) with the offset taken in to account.  Here's an example from the EWMH spec:

> Assume that the set up uses two monitors, one running at 1280x1024 and the other to the right running at 1024x768, with the top edge of the two physical displays aligned. If the panel wants to fill the entire bottom edge of the smaller display with a panel 50 pixels tall, it should set a bottom strut of 306, with bottom_start_x of 1280, and bottom_end_x of 2303.

One workaround is to make the workspaces top/bottom aligned when putting yabar at the top/bottom correspondingly.

**Comment by agzam** [comment]
You know what? It might be a bug in Yabar. To confirm I'm gonna need to try something else, maybe lemonbar or polybar. Sorry for delayed responses. Unfortunately I cannot use Linux at work.

**Comment by ch11ng** [comment]
@agzam Is this still a problem?

**Comment by agzam** [comment]
> Is this still a problem?

you know what, to be honest I'm not sure. I have stopped using yabar or any status bars - I realized Emacs already has everything I need - clock, battery, etc. The only thing I miss is the connection status widget, but I learned to live without it. 
So I don't know - maybe it's not an issue anymore. I'm going to close this ticket for now. And thank you very much for all the help and guidance, although I end up not using it. 
It may still be an issue for people who want to use statusbar apps with EXWM. I guess they can still comment here or re-open the issue
